modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/utils/ImplicitObjectUtils.java

```diff
@@ -81,7 +81,14 @@ public static ResolverSet buildImplicitResolverSet(ParameterizedModel parameteri
         .orElse(null);
   }
 
-  private static boolean canBeUsedImplicitly(ParameterizedModel parameterizedModel) {
+  /**
+   * Returns whether the given model can be used implicitly. That means that all of its parameters which are non optional have
+   * associated a default value.
+   * 
+   * @param parameterizedModel model
+   * @return whether the given model can be used implicitly or not.
+   */
+  public static boolean canBeUsedImplicitly(ParameterizedModel parameterizedModel) {
     return parameterizedModel.getAllParameterModels().stream().noneMatch(p -> p.isRequired() && p.getDefaultValue() == null);
```

Comment:
we should check for required only, having a default on the optional or not having it is irrelevant for implicit creation of the element. Also, a required with a non null default value should be an error in the model.